src/main/java/com/uid2/admin/vertx/service/OperatorKeyService.java

```diff
@@ -184,6 +187,13 @@ private void handleOperatorAdd(RoutingContext rc) {
                 return;
             }
 
+            Set<Role> roles = RequestUtil.getRoles(rc.queryParam("roles").get(0)) == null ? new HashSet<>() : RequestUtil.getRoles(rc.queryParam("roles").get(0));
```

Comment:
actually when i was testing my changes, i noticed that "rc.queryParam("roles").get(0)" will throw exception - what you want to check is actually "rc.queryParam("roles").isEmpty()" as queryParams return a list. If returned list is empty, that param doesn't exist so should create a new HashSet<>() instead. might want to fix this in other places within this file